rsmtool/configuration_parser.py

```diff
@@ -897,41 +969,31 @@ def load_normalize_and_validate_config_from_dict(self,
 
         Returns
         -------
-        config_obj : Configuration
-            A configuration object
-        """
-        self.load_config_from_dict(config_dict, configdir)
-        return self.normalize_validate_and_process_config(context=context)
-
-
-
-
-    def read_config_from_file(self, filepath):
-        """
-        Read the configuration file.
+        configuration : Configuration
+            A Configuration object containing the parameters in the
+            file that we instantiated the parser for.
+        """
+        _, extension = splitext(self._filename)
```

Comment:
Do we want to use Pathlib here since we are refactoring anyway?

Reply:
Note that `parse()` can only be used on an instantiated `ConfigurationParser` object and `__init__()` takes care of handling Paths. So, not sure the you would ever _not_ get a filename here.

Reply:
I meant: do we want to continue use `os` modules or switch to exclusively `pathlib` as per https://github.com/EducationalTestingService/rsmtool/issues/370 ?

Reply:
🤦‍♂. Yes, good idea :)